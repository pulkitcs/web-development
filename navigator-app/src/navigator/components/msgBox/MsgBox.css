.msg-box {
  margin-top: 20px;
  font-size: 14px;
  line-height: 30px;
  padding: 5px 10px;
  height: auto;
  background-color: #f4f4f4;
}

.msg-box h2{
  font-size: 16px;
  font-weight: normal;
  margin-bottom: 5px;
}

.status {
  text-transform: capitalize;
  font-size: 13px;
}

.status > .error {
  font-size: 13px;
}

.status p{
  margin: 5px 0 5px 0;
  line-height: 18px;
}

.status.msg {
  height: 25px;
  font-weight: bold;
}

.status.msg > .error{
  color: #ee3124;
  height: auto;
  line-height: 25px;
  margin-top: 0;
}

.status.msg > img,
.status.msg > span {
  display: inline-block;
  float: left;
  margin-right: 5px;
}

.clear {
  clear: both;
}