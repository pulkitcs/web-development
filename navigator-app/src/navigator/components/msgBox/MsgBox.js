import React from 'react';
import ERROR_MSG from '../../config/error';
import API_SETTINGS from '../../config/api';
import './MsgBox.css';

const errorMsg = Object.create(ERROR_MSG);
const apiMsg = Object.create(API_SETTINGS);

// Component to render the received API data
const InformationBox = ({routesData}) => (
    <div className="status">
      <p>Total Distance: <strong>{ routesData[0] } Km</strong></p>
      <p>Total Time: <strong>{ routesData[1] } Hrs</strong></p>
    </div>
  );

// Method to check error type and return relevent css class
const errorMessageClass = (msg) => errorMsg.API_SERVER_ERROR === msg || errorMsg.FAILED_TO_FETCH === msg ? 'error' : '';

// Container for error messages and information
const MsgBox = ({msg, routesData}) => {
  return (
    <div>
      { 
        (Boolean(msg) || routesData.length > 0 ) &&
          <div className = 'msg-box'>
            { 
              Boolean(msg) && msg !== apiMsg.MOCK_API.MSG_SUCCESS &&
              <div className="status msg">
                <div className={ errorMessageClass(msg) }>
                  { msg }
                </div>
              </div> 
            }
            { routesData.length > 0 && <InformationBox routesData={routesData} /> }
          </div>
      }
    </div>
  );
}

export default MsgBox;