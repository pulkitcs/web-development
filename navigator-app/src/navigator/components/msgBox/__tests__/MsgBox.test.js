import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import MsgBox from '../msgBox';

describe('<MsgBox /> component', () => {

  const text = 'Rendering';

  it('should match the snapshot', () => {
    const tree = renderer.create(<MsgBox msg={text} routesData={[1,2]}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the HTML Correctly', () => {
    const wrapper = mount(<MsgBox msg={text} routesData={[1,2]}/>);
    
    expect(wrapper.find('.status').length).toBe(2);
  })

  it('should render the HTML Correctly, when routesData [] is empty', () => {
    const wrapper = mount(<MsgBox msg={text} routesData={[]} />);
    
    expect(wrapper.find('.status').length).toBe(1);
    expect(wrapper.find('.status').text()).toBe(text);
  })
})