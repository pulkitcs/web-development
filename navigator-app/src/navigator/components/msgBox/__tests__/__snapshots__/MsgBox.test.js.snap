// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MsgBox /> component should match the snapshot 1`] = `
<div>
  <div
    className="msg-box"
  >
    <div
      className="status msg"
    >
      <div
        className=""
      >
        Rendering
      </div>
    </div>
    <div
      className="status"
    >
      <p>
        Total Distance: 
        <strong>
          1
           Km
        </strong>
      </p>
      <p>
        Total Time: 
        <strong>
          2
           Hrs
        </strong>
      </p>
    </div>
  </div>
</div>
`;
